distribution/docker/src/docker/Dockerfile

```diff
@@ -115,6 +115,78 @@ RUN sed -i -e 's/ES_DISTRIBUTION_TYPE=tar/ES_DISTRIBUTION_TYPE=docker/' bin/elas
     chmod 0775 bin config config/jvm.options.d data logs plugins && \\
     find config -type f -exec chmod 0664 {} +
 
+<% if (docker_base == "fips") { %>
+
+USER root
+
+# Add plugins infrastructure
+RUN mkdir -p /opt/plugins/archive
+RUN chmod -R 0555 /opt/plugins
+
+RUN mkdir -p /fips/libs
+COPY fips/libs/*.jar /fips/libs/
+
+COPY filebeat-${version}.tar.gz metricbeat-${version}.tar.gz /tmp/
+RUN set -eux ; \\
+    for beat in filebeat metricbeat ; do \\
+      if [ ! -s /tmp/\$beat-${version}.tar.gz ]; then \\
+        echo "/tmp/\$beat-${version}.tar.gz is empty - cannot uncompress" 2>&1 ; \\
+        exit 1 ; \\
+      fi ; \\
+      if ! tar tf /tmp/\$beat-${version}.tar.gz >/dev/null; then \\
+        echo "/tmp/\$beat-${version}.tar.gz is corrupt - cannot uncompress" 2>&1 ; \\
+        exit 1 ; \\
+      fi ; \\
+      mkdir -p /opt/\$beat ; \\
+      tar xf /tmp/\$beat-${version}.tar.gz -C /opt/\$beat --strip-components=1 ; \\
+    done
+
+COPY plugins/*.zip /opt/plugins/archive/
+
+RUN chown 1000:1000 /opt/plugins/archive/*
+RUN chmod 0444 /opt/plugins/archive/*
+
+COPY fips/resources/fips_java_oracle.security /usr/share/elasticsearch/config/fips_java.security
+COPY fips/resources/fips_java.policy /usr/share/elasticsearch/config/fips_java.policy
+
+WORKDIR /usr/share/elasticsearch/config
+# Add fips specific configuration
+RUN cat <<EOF >> /usr/share/elasticsearch/config/elasticsearch.yml
+xpack.security.autoconfiguration.enabled: false
+xpack.security.authc.api_key.cache.hash_algo: "ssha256"
+xpack.security.authc.reserved_realm.enabled: false
+xpack.security.authc.password_hashing.algorithm: pbkdf2_stretch
+xpack.security.enabled: true
+xpack.security.enrollment.enabled: false
+xpack.security.fips_mode.enabled: true
+xpack.security.fips_mode.required_providers: ["BCFIPS", "BCJSSE"]
+xpack.security.http.ssl.enabled: true
+xpack.security.http.ssl.key: node1.key
+xpack.security.http.ssl.certificate: node1.crt
+xpack.security.http.ssl.certificate_authorities: node1.crt
```

Comment:
these keys will work for netty; but not for BCJSSE in approved only mode.

Was this tested or solved?

Reply:
this has been removed